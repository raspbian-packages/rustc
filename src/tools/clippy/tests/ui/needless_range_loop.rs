fn calc_idx(i: usize) -> usize {
    (i + i + 20) % 4
}

fn main() {
    let ns = [2, 3, 5, 7];

    for i in 3..10 {
        println!("{}", ns[i]);
    }

    for i in 3..10 {
        println!("{}", ns[i % 4]);
    }

    for i in 3..10 {
        println!("{}", ns[i % ns.len()]);
    }

    for i in 3..10 {
        println!("{}", ns[calc_idx(i)]);
    }

    for i in 3..10 {
        println!("{}", ns[calc_idx(i) % 4]);
    }

    let mut ms = vec![1, 2, 3, 4, 5, 6];
    for i in 0..ms.len() {
        ms[i] *= 2;
    }
    assert_eq!(ms, vec![2, 4, 6, 8, 10, 12]);

    let mut ms = vec![1, 2, 3, 4, 5, 6];
    for i in 0..ms.len() {
        let x = &mut ms[i];
        *x *= 2;
    }
    assert_eq!(ms, vec![2, 4, 6, 8, 10, 12]);

    let g = vec![1, 2, 3, 4, 5, 6];
    let glen = g.len();
    for i in 0..glen {
        let x: u32 = g[i+1..].iter().sum();
        println!("{}", g[i] + x);
    }
    assert_eq!(g, vec![20, 18, 15, 11, 6, 0]);

    let mut g = vec![1, 2, 3, 4, 5, 6];
    let glen = g.len();
    for i in 0..glen {
        g[i] = g[i+1..].iter().sum();
    }
    assert_eq!(g, vec![20, 18, 15, 11, 6, 0]);
}
