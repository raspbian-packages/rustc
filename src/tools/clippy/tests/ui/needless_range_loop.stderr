error: the loop variable `i` is only used to index `ns`.
  --> $DIR/needless_range_loop.rs:8:5
   |
8  | /     for i in 3..10 {
9  | |         println!("{}", ns[i]);
10 | |     }
   | |_____^
   |
   = note: `-D needless-range-loop` implied by `-D warnings`
help: consider using an iterator
   |
8  |     for <item> in ns.iter().take(10).skip(3) {
   |

error: the loop variable `i` is only used to index `ms`.
  --> $DIR/needless_range_loop.rs:29:5
   |
29 | /     for i in 0..ms.len() {
30 | |         ms[i] *= 2;
31 | |     }
   | |_____^
   |
help: consider using an iterator
   |
29 |     for <item> in &mut ms {
   |

error: the loop variable `i` is only used to index `ms`.
  --> $DIR/needless_range_loop.rs:35:5
   |
35 | /     for i in 0..ms.len() {
36 | |         let x = &mut ms[i];
37 | |         *x *= 2;
38 | |     }
   | |_____^
   |
help: consider using an iterator
   |
35 |     for <item> in &mut ms {
   |

